### PR TITLE
[wasi-common]: Hello wiggle!

### PR DESCRIPTION
**Pull request by kubkon**
Hello `wiggle`!

This PR migrates [`wiggle`](https://github.com/kubkon/wiggle) to ~~`crates/wasi-common/wiggle`~~ `crates/wiggle` as a subdir of `wasmtime` monorepo. I've strived to preserve _all_ `wiggle`'s git history.

All I've done so far in this PR is:
1. migrate the history and put `wiggle` inside ~~`crates/wasi-common/wiggle`~~ `crates/wiggle`
2. fix `wiggle`'s tests (i.e., paths to respective test `witx` files)
3. strip `wiggle` of workspace status

I've left `wiggle`'s inner folder structure intact. What I mainly mean by this is that I've not moved `wiggle`'s subcrates `generate` and `runtime` up a level. I figured this can always be done at a later stage.

### CODE REVIEW

**Comment by pchickey** [comment]
Looks great, thank you. My one bit of feedback is we should put it at crates/wiggle - we put a lot of effort into making it usable outside of just wasi-common and putting it at the same level in the hierarchy will help communicate it’s standalone nature.

File: crates/wasi-common/wiggle/crates/generate/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+target
```
[the remark this answers is not in the document]
Comment (kubkon) [reply crates/wasi-common/wiggle/crates/generate/.gitignore]: Indeed they are, good catch!

**Comment by kubkon** [comment]
> Looks great, thank you. My one bit of feedback is we should put it at crates/wiggle - we put a lot of effort into making it usable outside of just wasi-common and putting it at the same level in the hierarchy will help communicate it’s standalone nature.

SGTM!

**Comment by kubkon** [comment]
@pchickey and done, moved up the tree!

**Pull request merged**